components/formats-bsd/src/loci/formats/in/JPEGReader.java

```diff
@@ -207,8 +211,38 @@ protected void initFile(String id) throws FormatException, IOException {
         }
 
         HashMap<String, String> tags = exif.getTags();
-        for (String tagName : tags.keySet()) {
-          addGlobalMeta(tagName, tags.get(tagName));
+        for (String tagName : tags.keySet()) {  
+          if (tagName.equals(exif.getTagName(ExifSubIFDDirectory.TAG_FOCAL_PLANE_X_RES))) {
+            Length sizeX = FormatTools.getPhysicalSizeX(Double.parseDouble(tags.get(tagName)));
+            String units = tags.get(exif.getTagName(ExifSubIFDDirectory.TAG_FOCAL_PLANE_UNIT));
+            if (units != null)
+            {
+              if (Integer.parseInt(units) == 2) { 
+                sizeX = FormatTools.getPhysicalSizeX(Double.parseDouble(tags.get(tagName)), UNITS.INCH);
+              }
+              else if (Integer.parseInt(units) == 3) { 
+                sizeX = FormatTools.getPhysicalSizeX(Double.parseDouble(tags.get(tagName)), UNITS.CM);
+              }
+            }
+            store.setPixelsPhysicalSizeX(sizeX, 0);
+          }
+          else if (tagName.equals(exif.getTagName(ExifSubIFDDirectory.TAG_FOCAL_PLANE_Y_RES))) {
+            Length sizeY = FormatTools.getPhysicalSizeY(Double.parseDouble(tags.get(tagName)));
+            String units = tags.get(exif.getTagName(ExifSubIFDDirectory.TAG_FOCAL_PLANE_UNIT));
+            if (units != null)
+            {
+              if (Integer.parseInt(units) == 2) { 
+                sizeY = FormatTools.getPhysicalSizeY(Double.parseDouble(tags.get(tagName)), UNITS.INCH);
+              }
+              else if (Integer.parseInt(units) == 3) { 
+                sizeY = FormatTools.getPhysicalSizeY(Double.parseDouble(tags.get(tagName)), UNITS.CM);
+              }
+            }
+            store.setPixelsPhysicalSizeX(sizeY, 0);
+          }
+          else {
+            addGlobalMeta(tagName, tags.get(tagName));
```

Comment:
Should every tag be added to the global metadata?  i.e. should this be removed from the else block to it's used unconditionally?